### PR TITLE
Avoid shaded guava imports in Presto on Spark codebase

### PR DESCRIPTION
**Pull request by arhimondr**
Should resolve the issue mentioned here: https://github.com/prestodb/presto/pull/16154#issuecomment-852342015

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: src/checkstyle/presto-checks.xml [hunk 1]
```diff
@@ -209,6 +209,9 @@
             <property name="illegalPkgs" value="jersey.repackaged" />
             <property name="illegalPkgs" value="jdk.nashorn.internal" />
             <property name="illegalPkgs" value="jdk.internal" />
+            <!-- Added to avoid accidental imports of the Spark shaded guava library that lives under the org.spark_project.guava package -->
+            <!-- It seems unlikely that anything would need to be imported from the org.spark_project package, as it seem to be a private or semi private API -->
+            <property name="illegalPkgs" value="org.spark_project" />
```
Comment (viczhang861) [inline src/checkstyle/presto-checks.xml:214]: it `seems` to be

**Review by viczhang861 — APPROVED** [review]

**Review by v-jizhang — APPROVED** [review]
LGTM, I've tested it works with Spark 2.4.8

**Comment by arhimondr** [comment]
@v-jizhang Could you please try Spark 3.1 again with this diff included?

**Comment by v-jizhang** [comment]
> @v-jizhang Could you please try Spark 3.1 again with this diff included?

Tested. It works!

**Comment by arhimondr** [comment]
@v-jizhang Currently we have e2e tests based on Docker for Spark 2.4: https://github.com/prestodb/presto/blob/master/presto-spark-testing/src/test/java/com/facebook/presto/spark/testing/TestPrestoSparkLauncherIntegrationSmokeTest.java

We should probably add an e2e test for Spark 3.1+ as well. If you are planning to adopt the technology with Spark 3.1+ I wonder if you would be interested in adding an e2e test for that environment to make sure we don't break the integration along the way?

**Comment by v-jizhang** [comment]
> @v-jizhang Currently we have e2e tests based on Docker for Spark 2.4: https://github.com/prestodb/presto/blob/master/presto-spark-testing/src/test/java/com/facebook/presto/spark/testing/TestPrestoSparkLauncherIntegrationSmokeTest.java
> 
> We should probably add an e2e test for Spark 3.1+ as well. If you are planning to adopt the technology with Spark 3.1+ I wonder if you would be interested in adding an e2e test for that environment to make sure we don't break the integration along the way?

Thanks. I'll add e2e tests for Spark 3.1

**Pull request merged**